### ISSUE TITLE
MPI I/O hints are not preserved after closing the file access property list

### ISSUE THREAD

**Issue by wkliao** [HDFGroup/hdf5#3025]
MPI Info object was populated by the underneath MPI-IO library in PR #1636.
However, the hints are not preserved after the file access property list is closed.
Calling `H5Fget_access_plist()` and `H5Pget_fapl_mpio()` returns MPI_INFO_NULL.
Below is a test program. It is tested using the develop branch and 1.14.1-2.

```
#include <stdio.h>
#include <stdlib.h>
#include <mpi.h>
#include <hdf5.h>

int main (int argc, char **argv)
{
    herr_t err;
    hid_t file_id, fapl_id;
    MPI_Info info, info_used;

    MPI_Init(&argc, &argv);
    MPI_Info_create(&info);
    MPI_Info_set(info, "romio_no_indep_rw", "true");
 
    fapl_id = H5Pcreate(H5P_FILE_ACCESS);
    H5Pset_fapl_mpio(fapl_id, MPI_COMM_WORLD, info);
    file_id = H5Fcreate("dummy.h5", H5F_ACC_TRUNC, H5P_DEFAULT, fapl_id);
    H5Pclose(fapl_id);

    fapl_id = H5Fget_access_plist(file_id);
    H5Pget_fapl_mpio(fapl_id, NULL, &info_used);

    if (info_used == MPI_INFO_NULL) printf("Error: info is MPI_INFO_NULL\n");

    H5Pclose(fapl_id);
    H5Fclose(file_id);
    MPI_Info_free(&info);

    MPI_Finalize();
    return 0;
}
```

**Comment by hyoklee** [comment]
@derobins merged #1636.

**Comment by fortnern** [comment]
Summary: 1636 changed the library to retrieve MPI Info from MPI after opening the file, which may be different from the Info passe to MPI. This retrieved info is, however, only stored on the FAPL, so it is lost when the FAPL is closed. We should either modify the library to cache this info on the file struct or modify H5Fget_access_plist to retrieve the Info from MPI
